tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/sdk/BallerinaSdkUtils.java

```diff
@@ -320,6 +320,34 @@ public static VirtualFile getSdkSrcDir(@NotNull Project project, @Nullable Modul
         });
     }
 
+    /**
+     * Searches for a ballerina project using outward recursion starting from the file directory, until the given root
+     * directory is found. Returns and empty string if unable to detect any ballerina project under the current intellij
+     * project source root.
+     */
+    public static String searchForBallerinaProjectRoot(String currentPath, String root) {
+
+        if (currentPath.equals(root) || currentPath.equals("") || root.equals("")) {
+            return "";
+        }
+        File currentDir = new File(currentPath);
+        File[] files = currentDir.listFiles();
+        if (files != null) {
+            for (File f : files) {
+                //skips the .ballerina folder in the user home directory.
+                if (f.isDirectory() && !f.getParentFile().getAbsolutePath().equals(System.getProperty("user.home")) && f
```

Comment:
I think it might be better to make the `.ballerina` a constant.

Reply:
+1. Added to `BallerinaConstants` as BALLERINA_PROJECT_CACHE_FOLDER_NAME